Fix nightly with some new deprecations

```
FAILED tests/arrays/test_datetime_array.py::test_properties - pandas.errors.Pandas4Warning: DatetimeArray.dayofweek is deprecated and will be removed in a future version. Use DatetimeArray.day_of_week instead.
FAILED tests/arrays/test_period.py::test_period_array_dayofweek - pandas.errors.Pandas4Warning: PeriodArray.dayofweek is deprecated and will be removed in a future version. Use PeriodArray.day_of_week instead.
FAILED tests/arrays/test_period.py::test_period_array_dayofyear - pandas.errors.Pandas4Warning: PeriodArray.dayofyear is deprecated and will be removed in a future version. Use PeriodArray.day_of_year instead.
FAILED tests/arrays/test_period.py::test_period_array_daysinmonth - pandas.errors.Pandas4Warning: PeriodArray.daysinmonth is deprecated and will be removed in a future version. Use PeriodArray.days_in_month instead.
FAILED tests/scalars/test_scalars.py::test_timestamp_properties - pandas.errors.Pandas4Warning: Timestamp.dayofweek is deprecated and will be removed in a future version. Use Timestamp.day_of_week instead.
FAILED tests/scalars/test_scalars.py::test_period_properties - pandas.errors.Pandas4Warning: Period.dayofweek is deprecated and will be removed in a future version. Use Period.day_of_week instead.
FAILED tests/series/test_agg.py::test_agg_complex - Failed: DID NOT WARN. No warnings of type (<class 'numpy.exceptions.ComplexWarning'>,) were emitted.
FAILED tests/test_timefuncs.py::test_series_dt_accessors - pandas.errors.Pandas4Warning: DatetimeIndex.dayofweek is deprecated and will be removed in a future version. Use DatetimeIndex.day_of_week instead.
FAILED tests/test_timefuncs.py::test_datetimeindex_accessors - pandas.errors.Pandas4Warning: DatetimeIndex.dayofweek is deprecated and will be removed in a future version. Use DatetimeIndex.day_of_week instead.
FAILED tests/test_timefuncs.py::test_periodindex_accessors - pandas.errors.Pandas4Warning: PeriodArray.dayofweek is deprecated and will be removed in a future version. Use PeriodArray.day_of_week instead.
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -36,7 +36,7 @@ numpy = ">=1.23.5"
 
 [tool.poetry.group.dev.dependencies]
 mypy = ">=1.19.1"
-pandas = "3.0.0"
+pandas = "3.0.1"
 pyarrow = ">=10.0.1"
 pytest = ">=8.4.2"
 pyright = ">=1.1.408"

--- a/tests/arrays/test_datetime_array.py
+++ b/tests/arrays/test_datetime_array.py
@@ -239,17 +239,23 @@ def test_properties() -> None:
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
         lower="3.0.0",
-        upper="3.1.0",
+        upper="3.2.0",
     ):
         check(assert_type(arr.dayofweek, np_1darray_int32), np_1darray_int32, np.int32)
-
-    check(assert_type(arr.weekday, np_1darray_int32), np_1darray_int32, np.int32)
 
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
         lower="3.0.0",
-        upper="3.1.0",
+        upper="3.2.0",
+    ):
+        check(assert_type(arr.weekday, np_1darray_int32), np_1darray_int32, np.int32)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.2.0",
     ):
         check(assert_type(arr.dayofyear, np_1darray_int32), np_1darray_int32, np.int32)
 
@@ -260,7 +266,7 @@ def test_properties() -> None:
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
         lower="3.0.0",
-        upper="3.1.0",
+        upper="3.2.0",
     ):
         check(
             assert_type(arr.daysinmonth, np_1darray_int32), np_1darray_int32, np.int32

--- a/tests/arrays/test_datetime_array.py
+++ b/tests/arrays/test_datetime_array.py
@@ -238,24 +238,24 @@ def test_properties() -> None:
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(arr.dayofweek, np_1darray_int32), np_1darray_int32, np.int32)
 
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(arr.weekday, np_1darray_int32), np_1darray_int32, np.int32)
 
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(arr.dayofyear, np_1darray_int32), np_1darray_int32, np.int32)
 
@@ -265,8 +265,8 @@ def test_properties() -> None:
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(
             assert_type(arr.daysinmonth, np_1darray_int32), np_1darray_int32, np.int32

--- a/tests/arrays/test_datetime_array.py
+++ b/tests/arrays/test_datetime_array.py
@@ -24,10 +24,12 @@ from pandas.core.arrays.string_ import BaseStringArray
 import pytest
 
 from pandas._libs.tslibs.nattype import NaTType
+from pandas.errors import Pandas4Warning
 
 from tests import (
     TYPE_CHECKING_INVALID_USAGE,
     check,
+    pytest_warns_bounded,
 )
 from tests._typing import (
     NumpyTimestampDtypeArg,
@@ -232,12 +234,38 @@ def test_properties() -> None:
     check(assert_type(arr.second, np_1darray_int32), np_1darray_int32, np.int32)
     check(assert_type(arr.microsecond, np_1darray_int32), np_1darray_int32, np.int32)
     check(assert_type(arr.nanosecond, np_1darray_int32), np_1darray_int32, np.int32)
-    check(assert_type(arr.dayofweek, np_1darray_int32), np_1darray_int32, np.int32)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.1.0",
+    ):
+        check(assert_type(arr.dayofweek, np_1darray_int32), np_1darray_int32, np.int32)
+
     check(assert_type(arr.weekday, np_1darray_int32), np_1darray_int32, np.int32)
-    check(assert_type(arr.dayofyear, np_1darray_int32), np_1darray_int32, np.int32)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.1.0",
+    ):
+        check(assert_type(arr.dayofyear, np_1darray_int32), np_1darray_int32, np.int32)
+
     check(assert_type(arr.quarter, np_1darray_int32), np_1darray_int32, np.int32)
     check(assert_type(arr.days_in_month, np_1darray_int32), np_1darray_int32, np.int32)
-    check(assert_type(arr.daysinmonth, np_1darray_int32), np_1darray_int32, np.int32)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.1.0",
+    ):
+        check(
+            assert_type(arr.daysinmonth, np_1darray_int32), np_1darray_int32, np.int32
+        )
+
     check(assert_type(arr.is_month_start, np_1darray_bool), np_1darray_bool, np.bool)
     check(assert_type(arr.is_month_end, np_1darray_bool), np_1darray_bool, np.bool)
     check(assert_type(arr.is_quarter_start, np_1darray_bool), np_1darray_bool, np.bool)

--- a/tests/arrays/test_period.py
+++ b/tests/arrays/test_period.py
@@ -16,8 +16,12 @@ from pandas.core.arrays.period import PeriodArray
 import pyarrow as pa
 
 from pandas._libs.tslibs.period import Period
+from pandas.errors import Pandas4Warning
 
-from tests import check
+from tests import (
+    check,
+    pytest_warns_bounded,
+)
 from tests._typing import (
     np_1darray_bool,
     np_1darray_int64,
@@ -166,7 +170,14 @@ def test_period_array_dayofweek() -> None:
     """Test dayofweek property for PeriodArray."""
     idx = PeriodIndex(["2020-01-01", "2020-01-02", "2020-01-03"], freq="D")
     arr = PeriodArray(idx)
-    check(assert_type(arr.dayofweek, np_1darray_int64), np_1darray_int64)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.1.0",
+    ):
+        check(assert_type(arr.dayofweek, np_1darray_int64), np_1darray_int64)
 
 
 def test_period_array_weekday() -> None:
@@ -180,7 +191,14 @@ def test_period_array_dayofyear() -> None:
     """Test dayofyear property for PeriodArray."""
     idx = PeriodIndex(["2020-01-01", "2020-06-15", "2020-12-31"], freq="D")
     arr = PeriodArray(idx)
-    check(assert_type(arr.dayofyear, np_1darray_int64), np_1darray_int64)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.1.0",
+    ):
+        check(assert_type(arr.dayofyear, np_1darray_int64), np_1darray_int64)
 
 
 def test_period_array_day_of_year() -> None:
@@ -215,7 +233,14 @@ def test_period_array_daysinmonth() -> None:
     """Test daysinmonth property for PeriodArray."""
     idx = PeriodIndex(["2020-01", "2020-02", "2020-03"], freq="M")
     arr = PeriodArray(idx)
-    check(assert_type(arr.daysinmonth, np_1darray_int64), np_1darray_int64)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.1.0",
+    ):
+        check(assert_type(arr.daysinmonth, np_1darray_int64), np_1darray_int64)
 
 
 def test_period_array_is_leap_year() -> None:

--- a/tests/arrays/test_period.py
+++ b/tests/arrays/test_period.py
@@ -174,8 +174,8 @@ def test_period_array_dayofweek() -> None:
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(arr.dayofweek, np_1darray_int64), np_1darray_int64)
 
@@ -188,8 +188,8 @@ def test_period_array_weekday() -> None:
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(arr.weekday, np_1darray_int64), np_1darray_int64)
 
@@ -202,8 +202,8 @@ def test_period_array_dayofyear() -> None:
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(arr.dayofyear, np_1darray_int64), np_1darray_int64)
 
@@ -244,8 +244,8 @@ def test_period_array_daysinmonth() -> None:
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(arr.daysinmonth, np_1darray_int64), np_1darray_int64)
 

--- a/tests/arrays/test_period.py
+++ b/tests/arrays/test_period.py
@@ -175,7 +175,7 @@ def test_period_array_dayofweek() -> None:
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
         lower="3.0.0",
-        upper="3.1.0",
+        upper="3.2.0",
     ):
         check(assert_type(arr.dayofweek, np_1darray_int64), np_1darray_int64)
 
@@ -184,7 +184,14 @@ def test_period_array_weekday() -> None:
     """Test weekday property for PeriodArray."""
     idx = PeriodIndex(["2020-01-01", "2020-01-02", "2020-01-03"], freq="D")
     arr = PeriodArray(idx)
-    check(assert_type(arr.weekday, np_1darray_int64), np_1darray_int64)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.2.0",
+    ):
+        check(assert_type(arr.weekday, np_1darray_int64), np_1darray_int64)
 
 
 def test_period_array_dayofyear() -> None:
@@ -196,7 +203,7 @@ def test_period_array_dayofyear() -> None:
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
         lower="3.0.0",
-        upper="3.1.0",
+        upper="3.2.0",
     ):
         check(assert_type(arr.dayofyear, np_1darray_int64), np_1darray_int64)
 
@@ -238,7 +245,7 @@ def test_period_array_daysinmonth() -> None:
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
         lower="3.0.0",
-        upper="3.1.0",
+        upper="3.2.0",
     ):
         check(assert_type(arr.daysinmonth, np_1darray_int64), np_1darray_int64)
 

--- a/tests/indexes/str/test_add.py
+++ b/tests/indexes/str/test_add.py
@@ -45,13 +45,13 @@ def test_add_py_sequence() -> None:
     if TYPE_CHECKING_INVALID_USAGE:
         _0 = left + i  # type: ignore[operator] # pyright: ignore[reportOperatorIssue,reportUnknownVariableType]
     check(assert_type(left + r0, "pd.Index[str]"), pd.Index, str)
-    with pytest_warns_bounded(Pandas4Warning, "Operation with tuple", lower="3"):
+    with pytest_warns_bounded(Pandas4Warning, "Operation with tuple", lower="3.0.99"):
         check(assert_type(left + r1, "pd.Index[str]"), pd.Index, str)
 
     if TYPE_CHECKING_INVALID_USAGE:
         _1 = i + left  # type: ignore[operator] # pyright: ignore[reportOperatorIssue,reportUnknownVariableType]
     check(assert_type(r0 + left, "pd.Index[str]"), pd.Index, str)
-    with pytest_warns_bounded(Pandas4Warning, "Operation with tuple", lower="3"):
+    with pytest_warns_bounded(Pandas4Warning, "Operation with tuple", lower="3.0.99"):
         check(assert_type(r1 + left, "pd.Index[str]"), pd.Index, str)
 
 

--- a/tests/scalars/test_scalars.py
+++ b/tests/scalars/test_scalars.py
@@ -1619,7 +1619,15 @@ def test_period_properties() -> None:
     check(assert_type(p.second, int), int)
     check(assert_type(p.start_time, pd.Timestamp), pd.Timestamp)
     check(assert_type(p.week, int), int)
-    check(assert_type(p.weekday, int), int)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.2.0",
+    ):
+        check(assert_type(p.weekday, int), int)
+
     check(assert_type(p.weekofyear, int), int)
     check(assert_type(p.year, int), int)
     check(assert_type(p.freq, BaseOffset), Day)

--- a/tests/scalars/test_scalars.py
+++ b/tests/scalars/test_scalars.py
@@ -996,10 +996,32 @@ def test_timestamp_properties() -> None:
     check(assert_type(ts.asm8, np.datetime64), np.datetime64)
     check(assert_type(ts.day_of_week, int), int)
     check(assert_type(ts.day_of_year, int), int)
-    check(assert_type(ts.dayofweek, int), int)
-    check(assert_type(ts.dayofyear, int), int)
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.1.0",
+    ):
+        check(assert_type(ts.dayofweek, int), int)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.1.0",
+    ):
+        check(assert_type(ts.dayofyear, int), int)
+
     check(assert_type(ts.days_in_month, int), int)
-    check(assert_type(ts.daysinmonth, int), int)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.1.0",
+    ):
+        check(assert_type(ts.daysinmonth, int), int)
+
     check(assert_type(ts.is_leap_year, bool), bool)
     check(assert_type(ts.is_month_end, bool), bool)
     check(assert_type(ts.is_month_start, bool), bool)
@@ -1559,10 +1581,33 @@ def test_period_properties() -> None:
     check(assert_type(p.day, int), int)
     check(assert_type(p.day_of_week, int), int)
     check(assert_type(p.day_of_year, int), int)
-    check(assert_type(p.dayofweek, int), int)
-    check(assert_type(p.dayofyear, int), int)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.1.0",
+    ):
+        check(assert_type(p.dayofweek, int), int)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.1.0",
+    ):
+        check(assert_type(p.dayofyear, int), int)
+
     check(assert_type(p.days_in_month, int), int)
-    check(assert_type(p.daysinmonth, int), int)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.1.0",
+    ):
+        check(assert_type(p.daysinmonth, int), int)
+
     check(assert_type(p.end_time, pd.Timestamp), pd.Timestamp)
     check(assert_type(p.freqstr, str), str)
     check(assert_type(p.hour, int), int)

--- a/tests/scalars/test_scalars.py
+++ b/tests/scalars/test_scalars.py
@@ -999,16 +999,16 @@ def test_timestamp_properties() -> None:
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.1.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(ts.dayofweek, int), int)
 
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.1.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(ts.dayofyear, int), int)
 
@@ -1017,8 +1017,8 @@ def test_timestamp_properties() -> None:
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.1.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(ts.daysinmonth, int), int)
 
@@ -1585,16 +1585,16 @@ def test_period_properties() -> None:
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.1.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(p.dayofweek, int), int)
 
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.1.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(p.dayofyear, int), int)
 
@@ -1603,8 +1603,8 @@ def test_period_properties() -> None:
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.1.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(p.daysinmonth, int), int)
 
@@ -1623,8 +1623,8 @@ def test_period_properties() -> None:
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(p.weekday, int), int)
 

--- a/tests/series/str/test_add.py
+++ b/tests/series/str/test_add.py
@@ -53,25 +53,43 @@ def test_add_py_sequence() -> None:
     if TYPE_CHECKING_INVALID_USAGE:
         _0 = left + i  # type: ignore[operator] # pyright: ignore[reportOperatorIssue,reportUnknownVariableType]
     check(assert_type(left + r0, "pd.Series[str]"), pd.Series, str)
-    with pytest_warns_bounded(Pandas4Warning, "Operation with tuple", lower="3"):
+    with pytest_warns_bounded(
+        Pandas4Warning, "Operation with tuple", lower="3.0.99", upper="3.1.99"
+    ):
         check(assert_type(left + r1, "pd.Series[str]"), pd.Series, str)
 
     if TYPE_CHECKING_INVALID_USAGE:
         _1 = i + left  # type: ignore[operator] # pyright: ignore[reportOperatorIssue,reportUnknownVariableType]
     check(assert_type(r0 + left, "pd.Series[str]"), pd.Series, str)
-    with pytest_warns_bounded(Pandas4Warning, "Operation with tuple", lower="3"):
+    with pytest_warns_bounded(
+        Pandas4Warning, "Operation with tuple", lower="3.0.99", upper="3.1.99"
+    ):
         check(assert_type(r1 + left, "pd.Series[str]"), pd.Series, str)
 
     if TYPE_CHECKING_INVALID_USAGE:
         left.add(i)  # type: ignore[arg-type] # pyright: ignore[reportArgumentType,reportCallIssue]
     check(assert_type(left.add(r0), "pd.Series[str]"), pd.Series, str)
-    with pytest_warns_bounded(Pandas4Warning, "Series.add with a tuple", lower="3"):
+
+    with (
+        pytest_warns_bounded(
+            Pandas4Warning,
+            "Series.add with a tuple is deprecated and will be treated as scalar-like in a future version",
+            lower="3.0.99",
+            upper="3.1.99",
+        ),
+    ):
         check(assert_type(left.add(r1), "pd.Series[str]"), pd.Series, str)
 
     if TYPE_CHECKING_INVALID_USAGE:
         left.radd(i)  # type: ignore[arg-type] # pyright: ignore[reportArgumentType,reportCallIssue]
     check(assert_type(left.radd(r0), "pd.Series[str]"), pd.Series, str)
-    with pytest_warns_bounded(Pandas4Warning, "Series.radd with a tuple", lower="3"):
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "Series.radd with a tuple is deprecated and will be treated as scalar-like in a future version",
+        lower="3.0.99",
+        upper="3.1.99",
+    ):
         check(assert_type(left.radd(r1), "pd.Series[str]"), pd.Series, str)
 
 

--- a/tests/series/test_agg.py
+++ b/tests/series/test_agg.py
@@ -60,14 +60,14 @@ def test_agg_complex() -> None:
     with pytest_warns_bounded(
         np.exceptions.ComplexWarning,
         r"Casting complex values to real discards the imaginary part",
-        upper="3.1.0",
+        upper="3.0.99",
     ):
         check(assert_type(series.std(), np.float64), np.float64)
 
     with pytest_warns_bounded(
         np.exceptions.ComplexWarning,
         r"Casting complex values to real discards the imaginary part",
-        upper="3.1.0",
+        upper="3.0.99",
     ):
         check(assert_type(series.var(), float), np.float64)
 

--- a/tests/series/test_agg.py
+++ b/tests/series/test_agg.py
@@ -53,16 +53,21 @@ def test_agg_complex() -> None:
     with pytest_warns_bounded(
         np.exceptions.ComplexWarning,
         r"Casting complex values to real discards the imaginary part",
+        upper="3.2.0",
     ):
         check(assert_type(series.median(), float), np.float64)
+
     with pytest_warns_bounded(
         np.exceptions.ComplexWarning,
         r"Casting complex values to real discards the imaginary part",
+        upper="3.1.0",
     ):
         check(assert_type(series.std(), np.float64), np.float64)
+
     with pytest_warns_bounded(
         np.exceptions.ComplexWarning,
         r"Casting complex values to real discards the imaginary part",
+        upper="3.1.0",
     ):
         check(assert_type(series.var(), float), np.float64)
 

--- a/tests/test_timefuncs.py
+++ b/tests/test_timefuncs.py
@@ -337,10 +337,26 @@ def test_series_dt_accessors() -> None:
     check(assert_type(s0.dt.second, "pd.Series[int]"), pd.Series, np.integer)
     check(assert_type(s0.dt.microsecond, "pd.Series[int]"), pd.Series, np.integer)
     check(assert_type(s0.dt.nanosecond, "pd.Series[int]"), pd.Series, np.integer)
-    check(assert_type(s0.dt.dayofweek, "pd.Series[int]"), pd.Series, np.integer)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.1.0",
+    ):
+        check(assert_type(s0.dt.dayofweek, "pd.Series[int]"), pd.Series, np.integer)
+
     check(assert_type(s0.dt.day_of_week, "pd.Series[int]"), pd.Series, np.integer)
     check(assert_type(s0.dt.weekday, "pd.Series[int]"), pd.Series, np.integer)
-    check(assert_type(s0.dt.dayofyear, "pd.Series[int]"), pd.Series, np.integer)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.1.0",
+    ):
+        check(assert_type(s0.dt.dayofyear, "pd.Series[int]"), pd.Series, np.integer)
+
     check(assert_type(s0.dt.day_of_year, "pd.Series[int]"), pd.Series, np.integer)
     check(assert_type(s0.dt.quarter, "pd.Series[int]"), pd.Series, np.integer)
     check(assert_type(s0.dt.is_month_start, "pd.Series[bool]"), pd.Series, np.bool_)
@@ -350,7 +366,15 @@ def test_series_dt_accessors() -> None:
     check(assert_type(s0.dt.is_year_start, "pd.Series[bool]"), pd.Series, np.bool_)
     check(assert_type(s0.dt.is_year_end, "pd.Series[bool]"), pd.Series, np.bool_)
     check(assert_type(s0.dt.is_leap_year, "pd.Series[bool]"), pd.Series, np.bool_)
-    check(assert_type(s0.dt.daysinmonth, "pd.Series[int]"), pd.Series, np.integer)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.1.0",
+    ):
+        check(assert_type(s0.dt.daysinmonth, "pd.Series[int]"), pd.Series, np.integer)
+
     check(assert_type(s0.dt.days_in_month, "pd.Series[int]"), pd.Series, np.integer)
     check(assert_type(s0.dt.tz, dt.tzinfo | None), type(None))
     check(assert_type(s0.dt.freq, str | None), str)
@@ -634,10 +658,26 @@ def test_datetimeindex_accessors() -> None:
     check(assert_type(i0.second, "pd.Index[int]"), pd.Index, np.int32)
     check(assert_type(i0.microsecond, "pd.Index[int]"), pd.Index, np.int32)
     check(assert_type(i0.nanosecond, "pd.Index[int]"), pd.Index, np.int32)
-    check(assert_type(i0.dayofweek, "pd.Index[int]"), pd.Index, np.int32)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.2.0",
+    ):
+        check(assert_type(i0.dayofweek, "pd.Index[int]"), pd.Index, np.int32)
+
     check(assert_type(i0.day_of_week, "pd.Index[int]"), pd.Index, np.int32)
     check(assert_type(i0.weekday, "pd.Index[int]"), pd.Index, np.int32)
-    check(assert_type(i0.dayofyear, "pd.Index[int]"), pd.Index, np.int32)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.2.0",
+    ):
+        check(assert_type(i0.dayofyear, "pd.Index[int]"), pd.Index, np.int32)
+
     check(assert_type(i0.day_of_year, "pd.Index[int]"), pd.Index, np.int32)
     check(assert_type(i0.quarter, "pd.Index[int]"), pd.Index, np.int32)
     check(assert_type(i0.is_month_start, np_1darray_bool), np_1darray_bool)
@@ -647,7 +687,15 @@ def test_datetimeindex_accessors() -> None:
     check(assert_type(i0.is_year_start, np_1darray_bool), np_1darray_bool)
     check(assert_type(i0.is_year_end, np_1darray_bool), np_1darray_bool)
     check(assert_type(i0.is_leap_year, np_1darray_bool), np_1darray_bool)
-    check(assert_type(i0.daysinmonth, "pd.Index[int]"), pd.Index, np.int32)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.2.0",
+    ):
+        check(assert_type(i0.daysinmonth, "pd.Index[int]"), pd.Index, np.int32)
+
     check(assert_type(i0.days_in_month, "pd.Index[int]"), pd.Index, np.int32)
     check(assert_type(i0.tz, dt.tzinfo | None), type(None))
     check(assert_type(i0.freq, BaseOffset | None), BaseOffset)
@@ -728,13 +776,37 @@ def test_periodindex_accessors() -> None:
     check(assert_type(i0.hour, "pd.Index[int]"), pd.Index, np.integer)
     check(assert_type(i0.minute, "pd.Index[int]"), pd.Index, np.integer)
     check(assert_type(i0.second, "pd.Index[int]"), pd.Index, np.integer)
-    check(assert_type(i0.dayofweek, "pd.Index[int]"), pd.Index, np.integer)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.2.0",
+    ):
+        check(assert_type(i0.dayofweek, "pd.Index[int]"), pd.Index, np.integer)
+
     check(assert_type(i0.day_of_week, "pd.Index[int]"), pd.Index, np.integer)
     check(assert_type(i0.weekday, "pd.Index[int]"), pd.Index, np.integer)
-    check(assert_type(i0.dayofyear, "pd.Index[int]"), pd.Index, np.integer)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.2.0",
+    ):
+        check(assert_type(i0.dayofyear, "pd.Index[int]"), pd.Index, np.integer)
+
     check(assert_type(i0.day_of_year, "pd.Index[int]"), pd.Index, np.integer)
     check(assert_type(i0.quarter, "pd.Index[int]"), pd.Index, np.integer)
-    check(assert_type(i0.daysinmonth, "pd.Index[int]"), pd.Index, np.integer)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.2.0",
+    ):
+        check(assert_type(i0.daysinmonth, "pd.Index[int]"), pd.Index, np.integer)
+
     check(assert_type(i0.days_in_month, "pd.Index[int]"), pd.Index, np.integer)
     check(assert_type(i0.freq, BaseOffset | None), BaseOffset)
     check(assert_type(i0.strftime("%Y"), pd.Index), pd.Index, str)

--- a/tests/test_timefuncs.py
+++ b/tests/test_timefuncs.py
@@ -341,8 +341,8 @@ def test_series_dt_accessors() -> None:
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(s0.dt.dayofweek, "pd.Series[int]"), pd.Series, np.integer)
 
@@ -351,16 +351,16 @@ def test_series_dt_accessors() -> None:
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(s0.dt.weekday, "pd.Series[int]"), pd.Series, np.integer)
 
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(s0.dt.dayofyear, "pd.Series[int]"), pd.Series, np.integer)
 
@@ -377,8 +377,8 @@ def test_series_dt_accessors() -> None:
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(s0.dt.daysinmonth, "pd.Series[int]"), pd.Series, np.integer)
 
@@ -669,8 +669,8 @@ def test_datetimeindex_accessors() -> None:
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(i0.dayofweek, "pd.Index[int]"), pd.Index, np.int32)
 
@@ -679,16 +679,16 @@ def test_datetimeindex_accessors() -> None:
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(i0.weekday, "pd.Index[int]"), pd.Index, np.int32)
 
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(i0.dayofyear, "pd.Index[int]"), pd.Index, np.int32)
 
@@ -705,8 +705,8 @@ def test_datetimeindex_accessors() -> None:
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(i0.daysinmonth, "pd.Index[int]"), pd.Index, np.int32)
 
@@ -794,8 +794,8 @@ def test_periodindex_accessors() -> None:
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(i0.dayofweek, "pd.Index[int]"), pd.Index, np.integer)
 
@@ -803,16 +803,16 @@ def test_periodindex_accessors() -> None:
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(i0.weekday, "pd.Index[int]"), pd.Index, np.integer)
 
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(i0.dayofyear, "pd.Index[int]"), pd.Index, np.integer)
 
@@ -822,8 +822,8 @@ def test_periodindex_accessors() -> None:
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
-        lower="3.0.0",
-        upper="3.2.0",
+        lower="3.0.99",
+        upper="3.1.99",
     ):
         check(assert_type(i0.daysinmonth, "pd.Index[int]"), pd.Index, np.integer)
 

--- a/tests/test_timefuncs.py
+++ b/tests/test_timefuncs.py
@@ -342,18 +342,25 @@ def test_series_dt_accessors() -> None:
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
         lower="3.0.0",
-        upper="3.1.0",
+        upper="3.2.0",
     ):
         check(assert_type(s0.dt.dayofweek, "pd.Series[int]"), pd.Series, np.integer)
 
     check(assert_type(s0.dt.day_of_week, "pd.Series[int]"), pd.Series, np.integer)
-    check(assert_type(s0.dt.weekday, "pd.Series[int]"), pd.Series, np.integer)
 
     with pytest_warns_bounded(
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
         lower="3.0.0",
-        upper="3.1.0",
+        upper="3.2.0",
+    ):
+        check(assert_type(s0.dt.weekday, "pd.Series[int]"), pd.Series, np.integer)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.2.0",
     ):
         check(assert_type(s0.dt.dayofyear, "pd.Series[int]"), pd.Series, np.integer)
 
@@ -371,7 +378,7 @@ def test_series_dt_accessors() -> None:
         Pandas4Warning,
         "is deprecated and will be removed in a future version.",
         lower="3.0.0",
-        upper="3.1.0",
+        upper="3.2.0",
     ):
         check(assert_type(s0.dt.daysinmonth, "pd.Series[int]"), pd.Series, np.integer)
 
@@ -668,7 +675,14 @@ def test_datetimeindex_accessors() -> None:
         check(assert_type(i0.dayofweek, "pd.Index[int]"), pd.Index, np.int32)
 
     check(assert_type(i0.day_of_week, "pd.Index[int]"), pd.Index, np.int32)
-    check(assert_type(i0.weekday, "pd.Index[int]"), pd.Index, np.int32)
+
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.2.0",
+    ):
+        check(assert_type(i0.weekday, "pd.Index[int]"), pd.Index, np.int32)
 
     with pytest_warns_bounded(
         Pandas4Warning,
@@ -786,7 +800,13 @@ def test_periodindex_accessors() -> None:
         check(assert_type(i0.dayofweek, "pd.Index[int]"), pd.Index, np.integer)
 
     check(assert_type(i0.day_of_week, "pd.Index[int]"), pd.Index, np.integer)
-    check(assert_type(i0.weekday, "pd.Index[int]"), pd.Index, np.integer)
+    with pytest_warns_bounded(
+        Pandas4Warning,
+        "is deprecated and will be removed in a future version.",
+        lower="3.0.0",
+        upper="3.2.0",
+    ):
+        check(assert_type(i0.weekday, "pd.Index[int]"), pd.Index, np.integer)
 
     with pytest_warns_bounded(
         Pandas4Warning,

